### PR TITLE
Adjustments for release pipeline

### PR DESCRIPTION
**Pull request by radtriste**
Many thanks for submitting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `KOGITO-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting master: `[0.9.x] KOGITO-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

**Comment by radtriste** [comment]
@Kevin-Mok Good for review&merge

**Comment by radtriste** [comment]
noticed some problems in jenkins-tests ... gonna look into it.

File: Jenkinsfile.deploy.new [hunk 1]
```diff
@@ -5,54 +5,76 @@ pipeline {
...
         string(name: 'PROJECT_VERSION', defaultValue: '', description: 'Optional if not RELEASE. If RELEASE, cannot be empty.')
     }
+
     environment {
         MAVEN_OPTS = '-Xms1024m -Xmx4g'
+
         BOT_CREDENTIALS_ID = "bsig-gh-bot"
+        GIT_AUTHOR_BOT = "bsig-gh-bot"
+
+        BOT_BRANCH_HASH = "${util.generateHash(10)}"
```
Comment (Kevin-Mok) [inline Jenkinsfile.deploy.new:42]: What's the purpose of the hash? Failed runs of the same release?

**Comment by radtriste** [comment]
@Kevin-Mok all yours for review :)

File: Jenkinsfile.promote [hunk 2]
```diff
@@ -88,15 +118,16 @@ pipeline {
...
-                    forcePushProtectedBranch("kogito-examples", "stable", "master")
+                script {
+                    dir('kogito-examples') {
+                        checkoutRepo('kogito-examples')
+                        mergeAndPush('kogito-examples', getDeployPrLink('kogito-examples'))
+                        tagLatest()
+
+
+                        githubscm.createBranch("stable")
+                        forcePushProtectedBranch('kogito-examples', "stable", "master")
```
Comment (Kevin-Mok) [inline Jenkinsfile.promote]:
```suggestion
                        githubscm.createBranch('stable')
                        forcePushProtectedBranch('kogito-examples', 'stable', 'master')
```

File: Jenkinsfile.promote [hunk 3]
```diff
@@ -255,40 +329,42 @@ void addNotIgnoredPoms() {
...
-        } else {
-            error "Couldn't update default branch."
+    withCredentials([usernamePassword(credentialsId: env.AUTHOR_CREDS_ID, usernameVariable: 'GIT_USERNAME', passwordVariable: 'GIT_PASSWORD')]) {
+        withCredentials([string(credentialsId: env.GITHUB_TOKEN_CREDS_ID, variable: 'GITHUB_TOKEN')]) {
+            // gh command from https://github.com/cli/cli/issues/929#issuecomment-629253585
+            def newDefaultBranch = sh(script: "../gh api -XPATCH 'repos/${getGitAuthor()}/${repo}' -f default_branch=${defaultBranch} | jq '.default_branch'", returnStdout: true).trim()
+            if (newDefaultBranch == "\"${defaultBranch}\"") {
+                echo "[INFO] ${getGitAuthor()}/${repo}'s default branch has been updated to ${newDefaultBranch}."
+            } else {
+                error "Couldn't update default branch."
```
Comment (Kevin-Mok) [inline Jenkinsfile.promote]:
```suggestion
                error 'Couldn't update default branch.'
```

**Review by Kevin-Mok — APPROVED** [review]

File: Jenkinsfile.deploy.new (hunk 1 above)
Comment (radtriste) [reply Jenkinsfile.deploy.new:42]: yep, don't want to have any conflict if an error happens but branch is already pushed and PR created. At least there is a "unique" branch name.
I did not like to use the BUILD_NUMBER and COMMIT_ID is not a good idea as the pipeline is taken from master

**Comment by radtriste** [comment]
@Kevin-Mok once you are good, you can merge it to yours so we can review your final PR ;)

**Pull request merged**